Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Commands/Commands.cs

```diff
@@ -157,19 +157,10 @@ public static Command DeleteAction(int actionMapIndex, string actionName)
                 var actionID = InputActionSerializationHelpers.GetId(action);
                 InputActionSerializationHelpers.DeleteActionAndBindings(actionMap, actionID);
                 state.serializedObject.ApplyModifiedProperties();
-                if (state.selectedActionIndex >= actionIndex)
-                    return SelectPrevAction(state, actionMap);
-                return state.SelectAction(state.selectedActionIndex);
-            };
-        }
 
-        private static InputActionsEditorState SelectPrevAction(InputActionsEditorState state, SerializedProperty actionMap)
-        {
-            var count = Selectors.GetActionCount(actionMap);
-            int index = -1;
-            if (count != null && count.Value > 0)
-                index = Math.Max(state.selectedActionIndex - 1, 0);
-            return state.SelectAction(index);
+                // ActionsTreeView will dispatch a separate command to select the previous Action
+                return state.With(selectedActionIndex: -1);
```

Comment:
How about just returning the state without changing the binding index? It will be overwritten by the ActionsTreeView anyway.